src/openforms/forms/admin/form.py

```diff
@@ -48,6 +48,21 @@ class FormStepInline(OrderedTabularInline):
     extra = 1
 
 
+class FormReachedSubmissionLimitListFilter(admin.SimpleListFilter):
+    title = _("has reached submission limit")
+    parameter_name = "submission_limit"
+
+    def lookups(self, request, model_admin):
+        return [("unavailable", "Unavailable for submission")]
+
+    def queryset(self, request, queryset):
+        if self.value() == "unavailable":
+            return queryset.filter(
+                ~Q(submission_maximum_allowed=None)
+                & Q(submission_maximum_allowed=F("submission_counter"))
+            )
```

Comment:
```suggestion
            return queryset.filter(
                submission_maximum_allowed__isnull=False,
                submission_counter__gte=F("submission_maximum_allowed"),
            )
```

* `isnull` is a bit more explicit *and* allows you to drop the `Q` wrapper
* race conditions happen and unless you set up a database constraint, at some point there will be situations where the submission counter has  value larger than `submission_maximum_allowed`

An example of the second case happening in an innocent way:

1. Admin user sets submission limit of 100
2. Form gets 75 submissions
3. Admin user changes submission limit to 50

Now you have 75 submissions which is more than 50 - the form is not available.

Reply:
So, reading this and the comment below for the serializer's field validation should I add a db constraint?
Or the solution is just to make the form unavailable?

Reply:
just make the form unavailable, the above situation is a valid case when the number of submissions could be higher than the limit.